core/module_graph.cc

```diff
@@ -128,22 +128,58 @@ bool ModuleGraph::RemoveEdge(const std::string &from, const std::string &to) {
   return UpdateTaskGraph();
 }
 
-bool ModuleGraph::AddModule(Module *m) {
+// Creates a module to the graph.
+Module *ModuleGraph::CreateModule(const ModuleBuilder &builder,
+                                  const std::string &module_name,
+                                  const google::protobuf::Any &arg,
+                                  pb_error_t *perr) {
+  Module *m =
```

Comment:
Maybe this could be a std::unique_ptr?

Reply:
Unfortunately, it is somewhat awkward to use `unique_ptr` here since the pointer is stored in separate data structures in the middle of the function. :(

Reply:
You could `.release` the `unique_ptr` but I agree that it gets awkward.

Reply:
No worries.